src/Raven.Server/Web/Authentication/AdminCertificatesHandler.cs

```diff
@@ -215,36 +218,73 @@ private static X509Certificate2 GetRootCertificate(X509Chain chain)
+                collection.Import(certBytes);
+
+                var first = true;
+                var collectionPrimaryKey = string.Empty;
+                
+                foreach (var x509Certificate in collection)
                 {
-                    // Implementation of SslStream AuthenticateAsServer is different in Linux. See RavenDB-8524
-                    // A list of allowed CAs is sent from the server to the client. The handshake will fail if the client's CA is not in that list. This list is taken from the root and certificate authority stores of the OS.
-                    // In this workaround we make sure that the CA (who signed the server cert, which in turn signed the client cert) is registered in one of the OS stores.
+                    var currentCertificate = new CertificateDefinition
+                    {
+                        Name = certificate.Name,
+                        Permissions = certificate.Permissions,
+                        SecurityClearance = certificate.SecurityClearance,
+                        Password = certificate.Password
+                    };
 
-                    using (var currentUserStore = new X509Store(StoreName.Root, StoreLocation.CurrentUser))
+                    if (PlatformDetails.RunningOnPosix)
                     {
-                        currentUserStore.Open(OpenFlags.ReadOnly);
-                        var userCerts = currentUserStore.Certificates.Find(X509FindType.FindByIssuerDistinguishedName, x509Certificate.IssuerName, true);
+                        // Implementation of SslStream AuthenticateAsServer is different in Linux. See RavenDB-8524
+                        // A list of allowed CAs is sent from the server to the client. The handshake will fail if the client's CA is not in that list. This list is taken from the root and certificate authority stores of the OS.
+                        // In this workaround we make sure that the CA (who signed the server cert, which in turn signed the client cert) is registered in one of the OS stores.
 
-                        if (userCerts.Contains(x509Certificate) == false)
-                            throw new InvalidOperationException($"Cannot save the client certificate '{certificate.Name}'. " +
-                                                                $"First, you must register the issuer certificate '{x509Certificate.IssuerName}' in the trusted root store, on the server machine." +
-                                                                "This step is required because you are using a self-signed certificate or one with unknown issuer.");
+                        using (var currentUserStore = new X509Store(StoreName.Root, StoreLocation.CurrentUser))
+                        {
+                            currentUserStore.Open(OpenFlags.ReadOnly);
+                            var userCerts = currentUserStore.Certificates.Find(X509FindType.FindByIssuerDistinguishedName, x509Certificate.IssuerName, true);
```

Comment:
Do we really have to register the cert itself, and not its CA?
Do we not need to check the `StoreLocation.LocalMachine` as well?

Reply:
Yes, need the CA here.
See RavenDB-8904, only checking root and certificate authority stores. Other local machine stores are not supported.